test/suggestion.cpp

```diff
@@ -187,4 +186,65 @@ namespace {
     std::vector<std::string> resultSet = getSuggestions(archive, "summer in", archive.getEntryCount());
     ASSERT_EQ(expectedResult, resultSet);
   }
-}
+
+    TEST(Suggestion, incrementalSearch) {
+    std::vector<std::string> titles = {
+                                        "The chocolate factory",
+                                        "The wolf of Shingashina",
+                                        "The wolf of Wall Street",
+                                        "The wolf among sheeps",
+                                        "The wolf of Wall Street Book" ,
+                                        "Hour of the wolf",
+                                       };
+
+    std::vector<std::string> resultSet, expectedResult;
+
+    TempZimArchive tza("testZim");
+    const zim::Archive archive = tza.createZimFromTitles(titles);
+
+    // "the"
+    resultSet = getSuggestions(archive, "the", archive.getEntryCount());
+    expectedResult = {
+                      "The chocolate factory",
+                      "Hour of the wolf",
+                      "The wolf among sheeps",
+                      "The wolf of Shingashina",
+                      "The wolf of Wall Street",
+                      "The wolf of Wall Street Book",
+                    };
+
+    ASSERT_EQ(expectedResult, resultSet);
+
+    // "the wolf"
+    resultSet = getSuggestions(archive, "the wolf", archive.getEntryCount());
+    expectedResult = {
+                      "Hour of the wolf",
+                      "The wolf among sheeps",
+                      "The wolf of Shingashina",
+                      "The wolf of Wall Street",
+                      "The wolf of Wall Street Book",
```

Comment:
We should have controls here with a title having only "wolf", to secure the stopwords are handled properly.